lib/bolt/inventory/group.rb

```diff
@@ -110,7 +106,7 @@ def data_merge(data1, data2)
 
       # Returns all nodes contained within the group, which includes nodes from subgroups.
       def node_names
-        @groups.inject(local_node_names) do |acc, g|
+        @_node_names ||= @groups.inject(local_node_names) do |acc, g|
```

Comment:
Or, since `local_node_names` isn't used anywhere else anymore, drop `local_node_names` completely.

Reply:
it's actually caching these sets not shortcircuiting that causes the big improvement.

Reply:
If we cache `@_node_names`, and we only call `local_node_names` while initializing it, then we don't also need to cache the result of `local_node_names`. So that one at least we could get rid of.

Reply:
The memory is pretty similar ~100MB while preparing then ~175MB when actually executing.

It seems like getting rid of this and getting rid of the shortcircuiting only slows it down by half. I'm not sure why I was seeing 4x improvement before. I'll leave it out for now.